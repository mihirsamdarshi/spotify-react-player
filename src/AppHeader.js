import React, { Component } from 'react'

import './AppHeader.css'

class AppHeader extends Component {
  render() {
    return (
      <div className="AppHeader">
        <h1>Giph Me!</h1>
      </div>
    )
  }
}

export default AppHeader
