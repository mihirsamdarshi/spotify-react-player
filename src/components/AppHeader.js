import React from 'react';
import Typography from '@material-ui/core/Typography';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import SettingsButton from './SettingsButton';

export default function AppHeader() {
    return (
        <div className="appHeader">
            <AppBar position="static" color="primary">
                <Toolbar>
                    <Typography variant="h6" className="title">
                        Applotify
                    </Typography>
                    <SettingsButton/>
                </Toolbar>
            </AppBar>
        </div>
    );
}
