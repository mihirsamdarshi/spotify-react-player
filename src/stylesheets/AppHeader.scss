.title {
  flex-grow: 1;
}

.MuiAppBar-root {
  background-color: black;
  color: pink;
  margin-bottom: 20px;
}

.settingsButton {
  float: right;
}
