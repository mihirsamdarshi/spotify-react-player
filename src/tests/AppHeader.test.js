import React from 'react';
import ReactDOM from 'react-dom';
import AppHeader from '../components/AppHeader';
import TestRenderer from 'react-test-renderer';

const { act } = TestRenderer;

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AppHeader />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should match the component', () => {
    const component = TestRenderer.create(<AppHeader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
