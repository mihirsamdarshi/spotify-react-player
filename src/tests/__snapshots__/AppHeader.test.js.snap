// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the component 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
  style={
    Object {
      "backgroundColor": "black",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <h6
      className="MuiTypography-root title MuiTypography-h6"
    >
      Applotify
    </h6>
  </div>
</header>
`;
